test/custom-parser.test.js

```diff
@@ -8,6 +8,29 @@ const sget = require('simple-get').concat
 const Fastify = require('..')
 
 const jsonParser = require('fast-json-body')
+// Adapted from https://github.com/delvedor/fast-json-body/blob/master/json-body.js
+function plainTextParser (request, callback) {
+  var body = ''
+  request.on('error', onError)
```

Comment:
for completeness, you need to call `request.setEncoding('utf8')`. I'm not sure why this is not failing.

Reply:
`request.setEncoding('utf8')` is not present in @delvedor source, but I'll add it.

Reply:
done.